File: linkedlist/insert_at_first.c
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int data;
    struct node * next;
};
void printlinkedlist(struct node * head){
    struct node *temp = head;
    while (temp!=NULL)
    {
        printf("%d ->",temp->data);
        temp = temp -> next;
    }
    printf("NULL\n");
}
struct node *insertatfirst(struct node*head ,int data){
    struct node*newnode = (struct node*)malloc(sizeof(struct node));
    newnode ->data =data;
    newnode ->next = head;
    head = newnode;
    return head;
}
int main(){
    struct node * head = (struct node * )malloc(sizeof(struct node));
    struct node * first = (struct node * )malloc(sizeof(struct node));
    struct node * second = (struct node * )malloc(sizeof(struct node));
    struct node * third = (struct node * )malloc(sizeof(struct node));
    head -> data = 20;
    head -> next = first;
    first -> data =30;
    first ->next = second;
    second ->data = 40;
    second -> next = third;
    third -> data = 50;
    third ->next = NULL;
    head = insertatfirst(head ,10);
    printlinkedlist(head);
    
    return 0 ;
}

File: linked  in java/insert_at_first.java
public class insert_at_first{
    class Node{
        int data;
        Node next;
        Node(int data){
            this.data=data;
            this.next=null;
        }
    }
     Node head = null;
    public void addFirst(int data){
        Node newnNode= new Node(data);
        if(head==null) {
            head=newnNode;
            return;
        }
        newnNode.next =head;
        head = newnNode;
    }
    public void print(){
        Node temp = head;
        while (temp!=null) {
            System.out.print(temp.data+"->");
            temp = temp.next;
        }
        System.out.println("null");
    }
    public static void main(String[] args) {
        insert_at_first list = new insert_at_first();
        list.addFirst(10);
        list.addFirst(20);
        list.addFirst(30);
        list.print();
    }
}